src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,549 @@
+			query.schoolname != ''
+		);
+		
+		const validData = (
+			['inclusive', 'exclusive'].includes(data.data.type) || (!data.data.type && !data.courseMetadataIds) &&
+			data.datasourceId != ''
+		);
+
+		if (!validData || !validQuery) {
+			return false;
+		}
+		
+		return [{
+			webuntisConfig: {
+				url: query.url,
+				schoolname: query.schoolname,
+				username: query.username,
+				password: query.password,
+			},
+			datasourceId: data.datasourceId,
+			datatype: data.data.type,
+			courseMetadataIds: data.courseMetadataIds,
+			dryrun: !("dryrun" in params) || (params.dryrun !== 'false' && params.dryrun !== false),
+		}];
+	}
+
+	/**
+	* @see {Syncer#steps}
+    *
+	* Perform either fetching from WebUntis and storing metadata into 'webuntisMetadata' or
+	* Creating actual schulcloud 'courses'.
+	*/
+	steps() {
+		this.logInfo(`Running WebUntis School Year Sync.\n`);
+		// this.logInfo(`Parameters `, this.data);
+
+		if (this.data.dryrun) {
+			return this.createMetaDataFromWebUntisSteps(this.data);
+		} else {
+			return this.createCoursesFromMetaDataAndWebUntisSteps(this.data);
+		}
+	}
+
+	/**
+	* 
+	*/
+	async createMetaDataFromWebUntisSteps(params) {
+		this.stats.success = false;
+		
+		const config = params.webuntisConfig;
+
+		const session = await this.login(config);
+		const metaData = await this.fetchMetaData(session, params);
+		await this.logout(session);
+		await this.storeMetadata(metaData, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	*/
+	async createCoursesFromMetaDataAndWebUntisSteps(params) {
+		// this.logger.debug(params);
+
+		this.stats.success = false;
+
+		const config = params.webuntisConfig;
+
+		const metaData = await this.obtainMetadata(params);
+		const session = await this.login(config);
+
+		const data = await this.fetchData(session, metaData);
+
+		await this.logout(session);
+
+		await this.createCourses(data, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	getImportConditionEvaluator(params) {
+		if (params.datatype === 'inclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
+			}
+			return ((id) => params.courseMetadataIds.includes(id));
+		}
+		if (params.datatype === 'exclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
+			}
+			return ((id) => !params.courseMetadataIds.includes(id));
+		}
+		throw new BadRequest('invalid datatype');
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	async obtainMetadata(params) {
+		const metaData = await this.app.service('webuntisMetadata').find({
+			query: { datasourceId: params.datasourceId },
+			paginate: false
+		});
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} config 
+	*/
+	async login(config) {
+		await super.login(config.url, config.schoolname, config.username, config.password);
+
+		return {
+			session: this.session,
+			rpc: this.rpc
+		};
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	*/
+	async fetchMetaData(session) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['session' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)